modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/AbstractPageMemoryIndexStorage.java

```diff
@@ -108,7 +109,7 @@ protected AbstractPageMemoryIndexStorage(
 
     @Override
     public @Nullable RowId getNextRowIdToBuild() {
-        return busy(() -> {
+        return busyNonRead(() -> {
```

Comment:
Why is this a `nonRead`?

Reply:
Because it's not a read of index data, it's related to some metadata.

I renamed the methods, is this better?

Reply:
Still quite confusing, please add javadocs at least

Reply:
Added javadocs